storage-proto/src/convert.rs

```diff
@@ -286,6 +286,20 @@ impl From<generated::Transaction> for VersionedTransaction {
     }
 }
 
+impl From<TransactionError> for generated::TransactionError {
+    fn from(value: TransactionError) -> Self {
+        let stored_error = Into::<StoredTransactionError>::into(value).0;
```

Comment:
nit: this might just be me, but I prefer using `from` instead of all the `Into`s
```suggestion
        let stored_error = StoredTransactionError::from(value).0;
```